### PR TITLE
keyring 23.13.1 (new formula)

### PR DESCRIPTION
**Pull request by PythonCoderAS**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I tried to make a test that would write a secret to a keychain and read it back, but this caused weird errors on my local machine that only happened whenever I ran it with `brew test`.

### CODE REVIEW

File: Formula/keyring.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    url "https://files.pythonhosted.org/packages/8e/b3/8b16a007184714f71157b1a71bbe632c5d66dd43bc8152b3c799b13881e1/zipp-3.11.0.tar.gz"
+    sha256 "a7a22e05929290a67401440b39690ae6563279bced5f314609d9d03798f56766"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    shell_output("#{bin}/keyring get https://example.com HomebrewTest2", 1)
```
Comment (SMillerDev) [inline Formula/keyring.rb:37]:
```suggestion
    assert_match "some text", shell_output("#{bin}/keyring get https://example.com HomebrewTest2", 1)
```

**Comment by cho-m** [comment]
This has a different Python dependency tree on Linux https://github.com/jaraco/keyring/blob/v23.13.1/setup.cfg#L23-L24
```
	SecretStorage>=3.2; sys_platform=="linux"
	jeepney>=0.4.2; sys_platform=="linux"
```

---

```console
$ pipgrip --tree SecretStorage jeepney
secretstorage (3.3.3)
├── cryptography>=2.0 (39.0.0)
│   └── cffi>=1.12 (1.15.1)
│       └── pycparser (2.21)
└── jeepney>=0.6 (0.8.0)
jeepney (0.8.0)
```

File: Formula/keyring.rb (hunk 1 above)
Comment (PythonCoderAS) [reply Formula/keyring.rb:37]: This doesn't work because when a value does not exist, then it returns no output and a statuscode of 1.

**Comment by PythonCoderAS** [comment]
How would I deal with the Linux situation? Can I just make this not compatible for linux?

**Comment by PythonCoderAS** [comment]
Bump

**Comment by chenrui333** [comment]
> How would I deal with the Linux situation? Can I just make this not compatible for linux?

I think what @cho-m referred to is to make separate dependency list for linux (which is what is shown in the `pipgrip` list)

File: Formula/keyring.rb (hunk 1 above)
Comment (dawidd6) [reply Formula/keyring.rb:55]: Then make it like this `assert_equal, "", shell_output...`.

Comment (SMillerDev) [reply Formula/keyring.rb:55]: `assert_empty` actually

**Review by dawidd6 — DISMISSED** [review]

**Comment by chenrui333** [comment]
> @chenrui333 How do I do that? I'm not sure how to make a separate list.

It would be mostly `on_linux do`, I guess we can also followup on it.

Something like this

```
  on_linux do
    depends_on "xxx"
    
    resource "xxx" do
    end
  end
```

**Review by chenrui333 — DISMISSED** [review]

**Comment by chenrui333** [comment]
squashed the commits for the merge

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@PythonCoderAS, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**